### PR TITLE
Use dataframe CSV write

### PR DESCRIPTION
**Pull request by galexite**
With previous RDD-based implementation:

```
real    6m12.129s
user    43m11.899s
sys     1m29.536s
```

With dataframe-based implementation:

```
real    3m27.518s
user    23m42.930s
sys     0m43.574s
```

Conducted locally with 20 million rows.

### CODE REVIEW

File: partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala [hunk 1]
```diff
@@ -21,8 +20,11 @@ object PartitionRawAis {
       .option("sep", "\t")
       .csv(config.inputPath)
       .selectExpr(selectClause: _*)
-      .rdd
-      .keyBy(row => (row.getAs[Int]("year"), row.getAs[Int]("month")))
-      .writeYearMonthPartitions()
+      .write
+      .partitionBy("year", "month")
```
Comment (JoelClemence) [inline partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala]: Does this output directories in the format 'year=X' and 'month=Y'. 

Also, the original filename is included in the directory structyre

**Comment by galexite** [comment]
I've had to merge in #76 :(

File: deployment/src/main/terraform/modules/data-query/raw_partitioned_ais_catalog_table.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module ais_raw_partitioned_catalog_table {
```
Comment (JoelClemence) [inline deployment/src/main/terraform/modules/data-query/raw_partitioned_ais_catalog_table.tf]: In future either:
a) change destination branch to the branch that is being merged first (i.e. #76) 
b) don't merge in until the upstream branch is in master (if you don't need functionality)

It just makes reviewing changes harder as there are changes from both PRs in one,

File: partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala [hunk 3]
```diff
@@ -14,15 +16,21 @@ object PartitionRawAis {
...
       .schema(Schema.AIS_SCHEMA)
       .option("sep", "\t")
       .csv(config.inputPath)
       .selectExpr(selectClause: _*)
-      .rdd
-      .keyBy(row => (row.getAs[Int]("year"), row.getAs[Int]("month")))
-      .writeYearMonthPartitions()
+      .withColumn("input_ais_data_file", input_file_name())
+      .write
+      .partitionBy("year", "month", "day")
```
Comment (JoelClemence) [inline partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala]: This is not going to work, as if another data file has data for the same day as a previous data file some of the previous data will be overwritten, you need something to declash them both either unique id or the file name.

File: deployment/src/main/terraform/modules/data-query/raw_partitioned_ais_catalog_table.tf (hunk 2 above)
Comment (JoelClemence) [reply deployment/src/main/terraform/modules/data-query/raw_partitioned_ais_catalog_table.tf]: Also you're going to have a ton of merge conflicts when #76 is merged.

File: partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala (hunk 3 above)
Comment (JoelClemence) [reply partition-raw-ais/src/main/scala/uk/gov/ukho/ais/partitioning/PartitionRawAis.scala]: Didn't understand the append save mode

**Pull request merged**